src/core/Akka.Tests/Actor/ActorLookupSpec.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Actor.Internal;
+using Akka.TestKit;
+using Akka.Util.Internal;
+using FluentAssertions;
+using Xunit;
+
+namespace Akka.Tests.Actor
+{
+    public class ActorLookupSpec : AkkaSpec
+    {
+        public sealed class Create
+        {
+            public Create(string child)
+            {
+                Child = child;
+            }
+
+            public string Child { get; private set; }
+        }
+
+        public interface IQuery { }
+
+        public sealed class LookupElems : IQuery
+        {
+            public LookupElems(IEnumerable<string> path)
+            {
+                Path = path;
+            }
+
+            public IEnumerable<string> Path { get; private set; }
+        }
+
+        public sealed class LookupString : IQuery
+        {
+            public LookupString(string path)
+            {
+                Path = path;
+            }
+
+            public string Path { get; private set; }
+        }
+
+        public sealed class LookupPath : IQuery
+        {
+            public LookupPath(ActorPath path)
+            {
+                Path = path;
+            }
+
+            public ActorPath Path { get; private set; }
+        }
+
+        public sealed class GetSender : IQuery
+        {
+            public GetSender(IActorRef to)
+            {
+                To = to;
+            }
+
+            public IActorRef To { get; private set; }
+        }
+
+        public static readonly Props P = Props.Create<Node>();
+
+        public class Node : ReceiveActor
+        {
+            private IActorRefProvider Provider { get { return Context.System.AsInstanceOf<ActorSystemImpl>().Provider; } }
+
+            public Node()
+            {
+                Receive<Create>(create => Sender.Tell(Context.ActorOf(P, create.Child)));
+                Receive<LookupElems>(elems => Sender.Tell(Provider.ResolveActorRef(new RootActorPath(Provider.DefaultAddress) / elems.Path)));
```

Comment:
Definitely not needed - they're trying to deprecate and discourage its use, plus it interferes with the UID system. Don't want to port anything that has no future if we can help it.